examples/mlperf/model_train.py

```diff
@@ -1,9 +1,210 @@
-from tinygrad.tensor import Tensor
-from tinygrad.helpers import getenv
+import functools
+import os
+import time
+from tqdm import tqdm
+
+from tinygrad import Device, GlobalCounters, Tensor, TinyJit, dtypes
+from tinygrad.helpers import getenv, BEAM, WINO
+from tinygrad.nn.state import get_parameters, get_state_dict, safe_load, safe_save
+
+from examples.mlperf.helpers import get_training_state, load_training_state
 
 def train_resnet():
-  # TODO: Resnet50-v1.5
-  pass
+  from extra.models import resnet
+  from examples.mlperf.dataloader import batch_load_resnet
+  from extra.datasets.imagenet import get_train_files, get_val_files
+  from examples.mlperf.lr_schedulers import PolynomialDecayWithWarmup
+  from examples.mlperf.initializers import Conv2dHeNormal, Linear
+  from examples.hlb_cifar10 import UnsyncedBatchNorm
+
+  config = {}
+  seed = config["seed"] = getenv("SEED", 42)
+  Tensor.manual_seed(seed)  # seed for weight initialization
+
+  GPUS = config["GPUS"] = [f"{Device.DEFAULT}:{i}" for i in range(getenv("GPUS", 1))]
+  print(f"Training on {GPUS}")
+  for x in GPUS: Device[x]
+
+  # ** model definition and initializers **
+  num_classes = 1000
+  resnet.Conv2d = Conv2dHeNormal
+  resnet.Linear = Linear
+  if not getenv("SYNCBN"): resnet.BatchNorm = functools.partial(UnsyncedBatchNorm, num_devices=len(GPUS))
+  model = resnet.ResNet50(num_classes)
+
+  # shard weights and initialize in order
+  for k, x in get_state_dict(model).items():
+    if not getenv("SYNCBN") and ("running_mean" in k or "running_var" in k):
+      x.realize().shard_(GPUS, axis=0)
+    else:
+      x.realize().to_(GPUS)
+  parameters = get_parameters(model)
+
+  # ** hyperparameters **
+  epochs            = config["epochs"]            = getenv("EPOCHS", 45)
+  BS                = config["BS"]                = getenv("BS", 104 * len(GPUS))  # fp32 GPUS<=6 7900xtx can fit BS=112
+  EVAL_BS           = config["EVAL_BS"]           = getenv("EVAL_BS", BS)
+  base_lr           = config["base_lr"]           = getenv("LR", 8.4 * (BS/2048))
+  lr_warmup_epochs  = config["lr_warmup_epochs"]  = 5
```

Comment:
`getenv("WARMUP_EPOCHS", 5)`

I was trying to match the nvidia hparams you mentioned. Have you tried it?